app/javascript/widget/components/Form/PhoneInput.vue

```diff
@@ -0,0 +1,314 @@
+        return (
+          name.toLowerCase().includes(search) ||
+          dial_code.toLowerCase().includes(search) ||
+          id.toLowerCase().includes(search)
+        );
+      });
+    },
+    activeCountry() {
+      if (this.activeCountryCode) {
+        return this.countries.find(
+          country => country.id === this.activeCountryCode
+        );
+      }
+      return '';
+    },
+  },
+  methods: {
+    setContextValue(code) {
+      // This function is used to set the context value.
+      // The context value is used to set the value of the phone number field in the pre-chat form.
+      this.context.model = `${code}${this.phoneNumber}`;
+    },
+    onChange(e) {
+      if (e.target.value === '') {
+        // This is to reset the active country code and dial code when the user clears the phone number field.
+        this.activeCountryCode = '';
+        this.activeDialCode = '';
+      }
+      this.phoneNumber = e.target.value;
+      // This function is used to set the context value when the user types in the phone number field.
+      this.setContextValue();
+    },
+    dropdownItem() {
+      // This function is used to get all the items in the dropdown.
+      return Array.from(
+        this.$refs.dropdown.querySelectorAll(
+          'div.country-dropdown div.country-dropdown--item'
+        )
+      );
+    },
+    focusedOrActiveItem(className) {
+      // This function is used to get the focused or active item in the dropdown.
+      return Array.from(
+        this.$refs.dropdown.querySelectorAll(
+          `div.country-dropdown div.country-dropdown--item.${className}`
+        )
+      );
+    },
+    onKeyDownHandler(e) {
+      // Here the code does is to check if the dropdown is open.
+      // If it is open, then it checks if the user has pressed the arrow down key or arrow up key.
+      // If the user has pressed the arrow down key, then it will increment the selectedIndex by 1.
+      // If the user has pressed the arrow up key, then it will decrement the selectedIndex by 1.
+      // If the user has pressed the enter key, then it will select the country.
+      const { showDropdown, filteredCountriesBySearch, onSelectCountry } = this;
+      const { selectedIndex } = this;
+
+      if (showDropdown) {
+        if (hasPressedArrowDownKey(e)) {
+          e.preventDefault();
+          this.selectedIndex = Math.min(
+            selectedIndex + 1,
+            filteredCountriesBySearch.length - 1
+          );
+          this.scrollToFocusedOrActiveItem(this.focusedOrActiveItem('focus'));
+        } else if (hasPressedArrowUpKey(e)) {
+          e.preventDefault();
+          this.selectedIndex = Math.max(selectedIndex - 1, 0);
+          this.scrollToFocusedOrActiveItem(this.focusedOrActiveItem('focus'));
+        } else if (isEnter(e)) {
+          e.preventDefault();
+          onSelectCountry(filteredCountriesBySearch[selectedIndex]);
+        }
+      }
```

Comment:
The key listener and the dropdown events are repeated from other component. Can we standardize this and make it a single component. 

@scmmishra Please provide a direction here.